Introduce .sbtextrasopts

sideeffffect opened the pull request:
along with `-sbt-extras-opts` and `$SBT_EXTRAS_OPTS`.
This will be useful for settings like `-sbt-launch-repo`.

fixes https://github.com/paulp/sbt-extras/issues/272

---
Note: I would find it preferable, if vanilla sbt ignored unrecognized parameters (as suggested in https://github.com/sbt/sbt/issues/5486 ), but it that won't happen we can merge this. Or we can merge this to have a workaround in the meantime.

dwijnand commented:
Thanks for the issue and the PR, @sideeffffect.  I'm not sure about it, even the suggestion that the sbt-launcher-package scripts should ignore anything it doesn't understand.

Going to sit and think about this one for a bit...

paulp commented:
It's aesthetically displeasing that their script is forked from this one, but now we're expected to adapt to them. They should not be viewing the contents of .sbtopts as anything under their control or part of their project. On that basis, they should definitely be ignoring unrecognized parameters.

That said, things are what they are. I would vote for .sbtxopts as the name though. I'd rather this whole project were built around sbtx as the name of the binary, also to avoid the collisions, but "sbt-extras" is not a good name for what has long been effectively a replacement for sbt, at the CLI level anyway.

sideeffffect commented:
I've renamed the file to `.sbtxopts`.
Also, in the latter commit, I've renamed the remaining stuff in my PR (env variable names, cmd-line switches, ...) to the sbt-x style -- I presumed that it would be better if it were the same.

dwijnand commented:
I think I prefer one-word sbtx/SBTX.  (Btw, still haven't given this PR as a whole enough thought, so it's just a side-comment.)

sideeffffect commented:
@dwijnand makes sense, I've amended the PR

take your time guys, please let me know once you make up your minds about this whole issue :+1:
